pkg/handlers/webhookrhobsreceiver.go

```diff
@@ -95,17 +96,71 @@ func (h *WebhookRHOBSReceiverHandler) processAMReceiver(d AMReceiverData, ctx co
 			continue
 		}
 
-		err = h.processAlert(alert, mfn)
-		if err != nil {
-			log.WithError(err).Error("a firing alert could not be successfully processed")
+		// Handle firing alerts
+		if alert.Status == string(model.AlertFiring) {
+			err = h.processFiringAlert(alert, mfn)
+			if err != nil {
+				log.WithError(err).Error("a firing alert could not be successfully processed")
+			}
+			continue
+		}
+
+		// Handle resolving alerts
+		if alert.Status == string(model.AlertResolved) {
+			err = h.processResolvedAlert(alert, mfn)
+			if err != nil {
+				log.WithError(err).Error("a resolving alert could not be successfully processed")
+			}
+			continue
 		}
 	}
+
 	return &AMReceiverResponse{Error: nil, Status: "ok", Code: http.StatusOK}
 }
 
-// processAlert handles the pre-check verification and sending of a notification for a particular alert
+// processResolvedAlert handles resolve notifications for a particular alert
+// currently only handles removing limited support
+func (h *WebhookRHOBSReceiverHandler) processResolvedAlert(alert template.Alert, mfn oav1alpha1.ManagedFleetNotification) error {
+	// Alert did not include a limited support request - NOOP.
+	if val, ok := alert.Labels[AMLabelLimitedSupportNotification]; !ok || val == "false" {
+		return nil
+	}
+
+	hcID := alert.Labels[AMLabelAlertHCID]
+	fn := mfn.Spec.FleetNotification
+	fnLimitedSupportReason := fn.NotificationMessage
+
+	activeLSReasons, err := h.ocm.GetLimitedSupportReasons(hcID)
+	if err != nil {
+		return fmt.Errorf("unable to get limited support reasons for cluster %s:, %w", hcID, err)
+	}
+
+	for _, reason := range activeLSReasons {
+		// If the reason matches the fleet notification LS reason, remove it
+		// TODO(Claudio): Find a way to make sure the removed LS was also posted by OA
+		if strings.Contains(reason.Details(), fnLimitedSupportReason) {
+			err := h.ocm.RemoveLimitedSupport(hcID, reason.ID())
+			if err != nil {
+				metrics.IncrementFailedLimitedSupportRemoved(fn.Name)
+				metrics.SetResponseMetricFailure("clusters_mgmt")
+				return fmt.Errorf("limited support reason with ID '%s' couldn't be removed for cluster %s, err: %w", reason.ID(), hcID, err)
+			}
+			metrics.IncrementLimitedSupportRemovedCount(fn.Name)
+		}
+	}
+	// Reset the metric if we got correct Response from OCM
+	metrics.ResetMetric(metrics.MetricResponseFailure)
+
+	// TODO(Claudio): should we add some kind of LimitedSupportRemoved state on the FleetNotificationRecordItem?
+	// Currently, we keep no trace of this other than logs.
+	// We could add a counter for sent limited supports.
```

Comment:
This is likely my biggest open question with my PR: currently, FleetNotificationRecordItem only has a counter for sent SLs (explicitly mentions service logs). 
I think we should rename this to `managed_notifications_fired_sent` and `managed_notification_resolves_sent`, that would keep it neutral when it comes to the type of notification (SL/LS).

Reply:
I don't have an opinion either way on this - probably will defer to @ravitri (as the SME on OA)

Reply:
I do like the idea of a generic `managed_notifications_fired_sent` metric, as it anticipates a broader set of future notifications that can be sent to customers.

Reply:
For the API: https://github.com/openshift/ocm-agent-operator/pull/106
As noted in that PR, I am unsure what the flow here is, as the api is needed for https://github.com/openshift/ocm-agent/pull/78, but the api also doesn't fit the currently deployed ocm-agent. Do we have some kind of circular dependency here?